android-components/components/feature/media/src/main/java/mozilla/components/feature/media/service/MediaSessionServiceDelegate.kt

```diff
@@ -180,10 +186,32 @@ internal class MediaSessionServiceDelegate(
     }
 
     @VisibleForTesting
-    internal fun startForeground(sessionState: SessionState) {
-        notificationScope?.launch {
+    @Suppress("TooGenericExceptionCaught")
+    internal fun startForeground(
+        sessionState: SessionState,
+        coroutineContext: CoroutineContext = EmptyCoroutineContext,
+    ) {
+        notificationScope?.launch(coroutineContext) {
             val notification = notificationHelper.create(sessionState, mediaSession)
-            service.startForeground(notificationId, notification)
+            try {
+                service.startForeground(notificationId, notification)
+            } catch (e: Exception) {
+                if (
+                    Build.VERSION.SDK_INT >= Build.VERSION_CODES.S &&
+                    e is ForegroundServiceStartNotAllowedException
+                ) {
+                    // We should not encounter this exception if `android:foregroundServiceType="mediaPlayback"`
+                    // is added to the service. The crash reporter loses the stack trace for this
+                    // exception so we want to be able to track this crash independently to ensure
+                    // this case is fixed and be able to determine if there are other cases where we
+                    // might be trying to start foreground services from the background.
+                    // https://bugzilla.mozilla.org/show_bug.cgi?id=1802620
+                    crashReporter.submitCaughtException(e)
```

Comment:
Will swallowing this exception cause any breakage in the user experience?

Reply:
@MatthewTighe I haven't noticed any breakage and I used the same approach PocketCasts uses here: https://github.com/Automattic/pocket-casts-android/blob/dfb5f8686d60b27adaad033977b5b772afbe826a/modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackService.kt#L259. That being said I can't say for sure 🤔

Reply:
Cool, I would expect keeping the app from crashing is highest priority here anyway - was just curious whether this had any side-effects like notifications disappearing unexpectedly.